### PR TITLE
Implementation of additional topVE toggles

### PR DESCRIPTION
**Pull request by robinpaul85**
## Description

Tentative implementation of additional topVE toggles

Added option to "filter extreme values" based on minimum (fpkm) count and toggle between methods for selecting variably expressed genes ("variance" and "interquartile region").  

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by creilly8 — DISMISSED** [review]

File: server/routes/termdb.topVariablyExpressedGenes.ts [hunk 1]
```diff
@@ -110,13 +110,22 @@ async function computeGenes4nativeDs(
...
+		filter_extreme_values = false
+	}
+
 	const input_json = {
 		input_file: matrixFile,
 		samples: samples.join(','),
-		filter_extreme_values: true,
-		num_genes: q.maxGenes,
-		param: 'var'
+		filter_extreme_values: filter_extreme_values,
```
Comment (xzhou82) [inline server/routes/termdb.topVariablyExpressedGenes.ts:124]: simplify as filter_ex_v: q.filter_ex_v ? true : false

File: server/routes/termdb.topVariablyExpressedGenes.ts [hunk 2]
```diff
@@ -110,13 +110,22 @@ async function computeGenes4nativeDs(
...
+		num_genes: q.maxGenes
 	}
+
+	if (q?.filter_type?.type) input_json['param'] = q.filter_type.type
```
Comment (xzhou82) [inline server/routes/termdb.topVariablyExpressedGenes.ts:128]: q is always present
replace with input_json.param=xx

File: server/routes/termdb.topVariablyExpressedGenes.ts (hunk 1 above)
Comment (robinpaul85) [reply server/routes/termdb.topVariablyExpressedGenes.ts:124]: The two statements are not equivalent. if (q.filter_extreme_values) is not defined (which is an optional variable) then the statement will be false. But filter_extreme_values should be "true" by default (in my opinion).

File: server/routes/termdb.topVariablyExpressedGenes.ts (hunk 2 above)
Comment (robinpaul85) [reply server/routes/termdb.topVariablyExpressedGenes.ts:128]: I have set default filter_type=iqr. q.filter_type is also an optional variable.

File: server/routes/termdb.topVariablyExpressedGenes.ts (hunk 1 above)
Comment (xzhou82) [reply server/routes/termdb.topVariablyExpressedGenes.ts:128]: then do `filter_extreme_values: q.filter_extreme_values == 1`

File: server/routes/termdb.topVariablyExpressedGenes.ts [hunk 3]
```diff
@@ -110,13 +110,27 @@ async function computeGenes4nativeDs(
...
 	const input_json = {
 		input_file: matrixFile,
 		samples: samples.join(','),
-		filter_extreme_values: true,
+		filter_extreme_values: filter_extreme_values,
 		num_genes: q.maxGenes,
-		param: 'var'
+		param: filter_type
 	}
+	console.log('input_json:', input_json)
```
Comment (xzhou82) [inline server/routes/termdb.topVariablyExpressedGenes.ts:132]: del

**Review by xzhou82 — APPROVED** [review]

**Review by creilly8 — APPROVED** [review]

**Pull request merged**